### ISSUE TITLE
Can't start Zebra on testnet after starting on mainnet

### ISSUE THREAD

**Issue by hdevalence** [ZcashFoundation/parity-zcash#53]
Building `zebra` from source and then running
```
./target/release/zebra import ~/.zcash/blocks/
./target/release/zebra --testnet
```
gives
```
Trying to open database with incompatible genesis block
```
instead of syncing the testnet.

**Comment by dconnolly** [comment]
Correct, this annoying UX. It should be smart enough to pick a different default db for testnet/regnet unless overridden by `--data-dir` or similar.

**Comment by hdevalence** [comment]
I'm actually not sure how to blow away or even find the local state, we should probably document that too.

**Comment by hdevalence** [comment]
If no location is specified, the zebra data is kept in a platform-dependent location selected by the `app_dirs` crate, e.g., `~/.local/share/zebra`. The database is opened by some functions in `zebra/util.rs`.  Personally I don't like `util` modules (since the API shouldn't require utility functions to use), so I'd like to refactor some of that code while fixing this.

I'm wondering if the best fix for this problem is to isolate chain-specific data into named subdirectories (e.g., `zebra/testnet/db` and `zebra/mainnet/db` instead of `zebra/db`). WDYT @dconnolly?

**Comment by dconnolly** [comment]
I love this idea. 💯

Those chain-specific paths could still be overridden by passing a preferred path to `--data-dir` or something. We don't currently have a zebra-specific general config file format, just the internal `Config` struct constructed from command line params, I wonder if it would be useful to have one for these sort of overrides.

**Comment by hdevalence** [comment]
Yeah, I'm going to start a branch with the following changes:

- [ ] rework the `Config` struct so that it can be created in multiple ways
- [ ] isolate the db storage by network name
- [ ] move db creation work out of `zebra/util.rs` and into the `db` module
